src/components/hover-visuals.js

```diff
@@ -9,31 +9,49 @@ AFRAME.registerComponent("hover-visuals", {
   },
   init() {
     // uniforms are set from the component responsible for loading the mesh.
+    this.handleLeftVisuals = this.handleLeftVisuals.bind(this);
+    this.handleRightVisuals = this.handleRightVisuals.bind(this);
     this.uniforms = null;
   },
   remove() {
     this.uniforms = null;
   },
   tick() {
-    if (!this.uniforms || !this.uniforms.size) return;
+    if (!this.uniforms || !this.uniforms.length) return;
 
+    // Push down loops to assist JIT:
```

Comment:
How come this helps? I don't understand JIT

Reply:
I believe since the shape of the uniform object is loop variable dependent before, but now has a function dedicated for each shape, it can JIT type-specific versions of the code easier now